### PR TITLE
make eagle3 be compatible with dp attention 

### PR DESCRIPTION
**Pull request by yizhang2077**
## Motivation

1. llama eagle3 be compatible with dp attention
2. make EAGLE3 be compatible with deepep when tp/dp = 1:1 (if tp/dp !=1, we need transfer embedding from scatter mode to dp attention full mode)
3. change draft model logits size from `vocab_size` to `draft_vocab_size`
## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.ai/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.ai/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.ai/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.ai/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.ai/developer_guide/contribution_guide.html#benchmark-the-speed).

### CODE REVIEW

**Review by ispobock — COMMENTED** [review]
@yizhang2077 Could you add unit test for this change?

File: python/sglang/srt/models/llama_eagle3.py [hunk 1]
```diff
@@ -124,6 +254,7 @@ def __init__(
             config.vocab_size,
             config.hidden_size,
             prefix=add_prefix("embed_tokens", prefix),
+            enable_tp=not is_dp_attention_enabled(),
```
Comment (ispobock) [inline python/sglang/srt/models/llama_eagle3.py:166]: There are some points to discuss:
1. Should we also enable DP attention for the draft model when it’s enabled for the target model? Do we need an option for draft model to enable the DP attn?
2. If DP attention is enabled for the draft model, both target and draft must be in the same DP rank, otherwise KV cache may be inconsistent.
3. If enabled, how do performance and accuracy compare with TP?

Comment (yizhang2077) [reply python/sglang/srt/models/llama_eagle3.py:166]: 1. draft model and target model shares the same server_args and tp/attn_tp communication groups, so we do not need to add extra parameter for enable dp attention.
2. since 1, we just need to use the same tp/attn_tp group then we can keep kv cache consistent
